llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -4233,9 +4231,6 @@ bool InstCombinerImpl::transformConstExprCastCall(CallBase &Call) {
 
   AttributeSet FnAttrs = CallerPAL.getFnAttrs();
 
-  if (NewRetTy->isVoidTy())
-    Caller->setName("");   // Void type should not have a name.
```

Comment:
I think this bit might still be needed? Can you please add another function to your test that does `%res = call i32 @foo()` but does not use the result? Just want to make sure this doesn't hit some void name assertion.

Reply:
Yes this would necessary now after we have tightened the condition above to skip transformation only when return value has uses.